pkg/provider/provider.go

```diff
@@ -0,0 +1,604 @@
+					Weight: getIntRef(backend.Weight),
+				}
+			}
+
+			entrypoint, err := p.buildHTTPEntrypoint(portID)
+			if err != nil {
+				p.logger.Errorf("Unable to build HTTP entrypoint for Service %s/%s and portID %d: %v", ts.Service.Namespace, ts.Service.Name, portID, err)
+				continue
+			}
+
+			svcKey := getServiceKeyFromTrafficSplit(ts, svcPort.Port)
+			cfg.HTTP.Services[svcKey] = buildHTTPServiceFromTrafficSplit(backendSvcs)
+
+			directRtrKey := getRouterKeyFromTrafficSplitDirect(ts, svcPort.Port)
+			cfg.HTTP.Routers[directRtrKey] = buildHTTPRouter(rule, entrypoint, rtrMiddlewares, svcKey, priorityTrafficSplit)
+
+			// If the ServiceTrafficSplit is a backend of at least one TrafficSplit we need an additional router with
+			// a whitelist middleware which whitelists based on the X-Forwarded-For header instead of on the RemoteAddr value.
+			if len(ts.Service.BackendOf) > 0 && p.acl {
+				whitelistIndirect := buildWhitelistMiddlewareFromTrafficSplitIndirect(ts)
+				whitelistIndirectKey := getWhitelistMiddlewareKeyFromTrafficSplitIndirect(ts)
+				cfg.HTTP.Middlewares[whitelistIndirectKey] = whitelistIndirect
+
+				rule = buildHTTPRuleFromTrafficSplitIndirect(ts)
+				rtrMiddlewaresindirect := addToSliceCopy(middlewares, whitelistIndirectKey)
+
+				indirectRtrKey := getRouterKeyFromTrafficSplitIndirect(ts, svcPort.Port)
+				cfg.HTTP.Routers[indirectRtrKey] = buildHTTPRouter(rule, entrypoint, rtrMiddlewaresindirect, svcKey, priorityTrafficTargetIndirect)
+			}
+		}
+	case k8s.ServiceTypeTCP:
+		tcpRule := buildTCPRouterRule()
+
+		for _, svcPort := range ts.Service.Ports {
+			backendSvcs := make([]dynamic.TCPWRRService, len(ts.Backends))
+
+			for i, backend := range ts.Backends {
+				backendSvcKey := getServiceKeyFromTrafficSplitBackend(ts, svcPort.Port, backend)
+				cfg.TCP.Services[backendSvcKey] = buildTCPSplitTrafficBackendService(backend, svcPort.TargetPort.IntVal)
+				backendSvcs[i] = dynamic.TCPWRRService{
+					Name:   backendSvcKey,
+					Weight: getIntRef(backend.Weight),
+				}
+			}
+
+			entrypoint, err := p.buildTCPEntrypoint(ts.Service, svcPort.Port)
+			if err != nil {
+				p.logger.Errorf("Unable to build TCP entrypoint for Service %s/%s and port %d: %v", ts.Service.Namespace, ts.Service.Name, svcPort.Port, err)
+				continue
+			}
+
+			key := getServiceRouterKeyFromService(ts.Service, svcPort.Port)
+			cfg.TCP.Services[key] = buildTCPServiceFromTrafficSplit(backendSvcs)
+			cfg.TCP.Routers[key] = buildTCPRouter(tcpRule, entrypoint, key)
+		}
+
+	default:
+		return fmt.Errorf("unknown traffic-type %q", trafficType)
+	}
+
+	return nil
+}
+
+func (p *Provider) buildBlockAllRouters(cfg *dynamic.Configuration, svc *topology.Service) {
+	rule := buildHTTPRuleFromService(svc)
+
+	for portID, svcPort := range svc.Ports {
+		entrypoint, err := p.buildHTTPEntrypoint(portID)
+		if err != nil {
+			p.logger.Errorf("unable to build HTTP entrypoint for Service %s/%s and portID %d: %w", svc.Namespace, svc.Name, portID, err)
```

Comment:
```suggestion
			p.logger.Errorf("unable to build HTTP entrypoint for Service %s/%s and port %d: %w", svc.Namespace, svc.Name, portID, err)
```

Reply:
We should probably rename `portID 5` into `port number 5` but definitely not `port`. This value represents the ID of the port in the list of ports.

Reply:
I will do what you suggests but instead of using the `portID` variable, I will use the `svcPort.Port`. `portID` is really an implementation details that woudn't bring a useful information to the user.